service/security/test-steps/src/main/java/org/eclipse/kapua/service/authorization/steps/AuthorizationServiceSteps.java

```diff
@@ -2534,4 +2535,162 @@ public void iCreateTheAccessInfoEntities() throws Exception {
+
+        primeException();
+        try {
+            assertEquals(roleName, role.getName());
+            assertEquals(roleDescription, role.getDescription());
+            KapuaId roleId = role.getId();
+            roleService.delete(role.getScopeId(), roleId);
+        } catch (KapuaException ex) {
+            verifyException(ex);
+        }
+    }
+
+    @Then("^I search for a role with description \"([^\"]*)\"$")
+    public void iFindARoleWithDescription(String roleDesc) throws Throwable {
+        try {
+            primeException();
+            RoleQuery roleQuery = roleFactory.newQuery(getCurrentScopeId());
+            roleQuery.setPredicate(roleQuery.attributePredicate(RoleAttributes.DESCRIPTION, roleDesc, AttributePredicate.Operator.EQUAL));
+
+            stepData.remove("RoleListResult");
+            stepData.remove("RoleFound");
+            RoleListResult roleListResult = roleService.query(roleQuery);
+            stepData.put("RoleListResult", roleListResult);
+            stepData.put("RoleFound", roleListResult.getFirstItem());
+        } catch (KapuaException ke) {
+            verifyException(ke);
+        }
+    }
+
+    @Given("^I try to create roles with invalid characters in name$")
+    public void iTryToCreateRolesWithInvalidCharactersInName() throws Exception {
+        RoleCreator roleCreator = roleFactory.newCreator(SYS_SCOPE_ID);
+        String invalidCharacters = "!\"#$%&'()=»Ç>:;<.,⁄@‹›€*ı–°·‚±Œ„‰?“‘”’ÉØ∏{}|ÆæÒÔÓÌÏÎÍÅ«";
+        for (int i = 0; i < invalidCharacters.length(); i++) {
+            String roleName = "roleName" + invalidCharacters.charAt(i);
+            roleCreator.setName(roleName);
+
+            try {
+                primeException();
+                stepData.remove("Role");
+                Role role = roleService.create(roleCreator);
+                stepData.put("Role", role);
+            } catch (KapuaException ex) {
+                verifyException(ex);
+            }
+        }
+    }
+
+    @Given("^I try to create roles with invalid characters in description$")
+    public void iTryToCreateRolesWithInvalidCharactersInDescription() throws Exception {
+        RoleCreator roleCreator = roleFactory.newCreator(SYS_SCOPE_ID);
+        String invalidCharacters = "!\"#$%&'()=»Ç>:;<.,⁄@‹›€*ı–°·‚±Œ„‰?“‘”’ÉØ∏{}|ÆæÒÔÓÌÏÎÍÅ«";
+        for (int i = 0; i < invalidCharacters.length(); i++) {
+            String roleDescription = "roleDescription" + invalidCharacters.charAt(i);
+            roleCreator.setDescription(roleDescription);
+            roleCreator.setName("roleName"+i);
+
+            try {
+                primeException();
+                stepData.remove("Role");
+                Role role = roleService.create(roleCreator);
+                stepData.put("Role", role);
+            } catch (KapuaException ex) {
+                verifyException(ex);
+            }
+        }
+    }
+
+    @Then("^I update the role name with special characters$")
```

Comment:
Please change step name to: " update the role name with special characters ""#$%&'(‹›€ı–......."